### PR TITLE
Changes to hyperparameter grid and apply increasing batch size when training

### PR DESCRIPTION
**Pull request by gregheymans**
On my dataset (~1000 data), and using my version of fit_preset(), I obtain a 8-9% improvement (on the MAE and RMSE) on a 5-fold test set. Moreover, my function is faster because I compare 8 models (instead of 16 when using the last version of modnet).
Therefore, I would love to see the performance of my fit_preset function on the matbench. Indeed, on my dataset, my model is faster and gives better results. I would then be very curious to see if this performance could be extrapolate to the Matbench.

### CODE REVIEW

**Comment by ml-evs** [comment]
Hi @gregheymans, fit_preset has changed considerably in our recent work (#23) and now the grid search adapts to the number of features and samples. I'm not sure the approach here will work well across all of matbench.

What benefit did you see from restarting the fit procedure 4 times? The Adam learning rates will be reset each time so you may occasionally be able to bubble out of local minimum, but I think trusting Adam to handle the learning rates would still be preferable? If you see a performance increase from this then we might consider changing the beta_i parameters of Adam instead of manually retraining https://keras.io/api/optimizers/adam/

**Comment by gregheymans** [comment]
Hi @ml-evs ! I saw you did many changes to the fit_preset function, and I still need to have a look on it though. ;)
I am also surprised to see that with my model, I obtained better results than the one you just added on the GitHub. In order to be sure I am not overfitting, I need to use my fit_preset model on another dataset. I actually do have a completely different dataset I used for a different project. I'll compare the results obtained from this dataset for ours two fit_preset functions. If I still get better results with my fit_preset model, I'll come back to you. :)
By the way, the four fits of 250 epochs, when increasing the batch size, improved significantly my results (here is the paper from which I got this idea: https://openreview.net/pdf?id=B1Yy1BxCZ). Did you try the "4 fits technique" on the Matbench?
I actually keep the callbacks.
Best,
Greg

**Comment by ml-evs** [comment]
> I am also surprised to see that with my model, I obtained better results than the one you just added on the GitHub.

Ah great, we should definitely check this out then! It may be that you are getting (un)lucky with the grid search for your particular dataset, we also found a few matbench cases where we could improve performance by tuning the grid but it comes at the expense of performance on other datasets.

> By the way, the four fits of 250 epochs, when increasing the batch size, improved significantly my results (here is the paper from which I got this idea: https://openreview.net/pdf?id=B1Yy1BxCZ). Did you try the "4 fits technique" on the Matbench?

I think @ppdebreuck tried it (though with 2 fits); my understanding of that paper is that it can accelerate training by requiring fewer epochs but shouldn't necessarily improve final performance; not sure how that works when you are fixing the number of epochs each time?

**Comment by ppdebreuck** [comment]
Before making conclusions on what works best, a thorough test on matbench is needed. We could maybe (at a later stage, priority is to release v0.9 as used in the paper) extend the presets, if easily transferable and performing better on matbench. But it should be consistent. We left out some callbacks for specific reasons. I suggest to plan a meeting about this and the GA later this week. :)

**Comment by ml-evs** [comment]
Sounds good; we could even consider adding the increasing batch-size during training as a hyperparameter to search over...

**Comment by gregheymans** [comment]
Yeah, good idea. It won’t be a bad idea to try it !
Greg

**Comment by gregheymans** [comment]
I compared four fits of 250 epochs with one of 1000 epochs and observed an
improvement. I’ll double check this with another dataset to see if this
enhancement stays.
Cheers,
Greg

**Comment by ppdebreuck** [comment]
outdated
